src/main/java/step4/CarName.java

```diff
@@ -0,0 +1,48 @@
+package step4;
+
+public class CarName {
+
+    private String name;
+
+    public CarName(String name) {
+        validate(name);
+        this.name = name;
+    }
+
+    private void validate(String name) {
+        validateNotNull(name);
+        validateNotEmpty(name);
+        validateLengthShorterOrEqualThan5(name);
+    }
+
+    private void validateNotNull(String name) {
+        if (name == null) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private void validateNotEmpty(String name) {
+        if (name.trim().length() == 0) {
+            throw new IllegalArgumentException();
+        }
+    }
```

Comment:
피드백 잘 적용해주셨군요! 

name값이 없는 경우에 체크하는 로직이여서 validateNotNull과 validateNotEmpty은 같은 검증이라고 판단이 됩니다 ~ 
제가 앞전 리뷰에서 헷갈리게 전달 드렸다면 죄송하네요 🥲
아래와 같이 구현해주셔도 될 것 같습니다 🙏🏻
한가지 더 전달드리자면 예외처리 메세지는 최대한 많은 정보를 제공해주는 것이 좋습니다~ 
현재 구조에서 어렵다면 커스텀 익셉션 사용해보시는것도 추천드립니다! 
```suggestion
    private void validateNotNull(String name) {
        if (name == null || name.trim().length() == 0) {
            throw new IllegalArgumentException(");
        }
    }
```